src/lib/system-info.ts

```diff
@@ -0,0 +1,114 @@
+import * as systeminformation from 'systeminformation';
+import * as osUtils from 'os-utils';
+import { fs, child_process } from 'mz';
+
+export function getCpuUsage() {
+	return new Promise((resolve) => {
+		osUtils.cpuUsage((percent) => {
+			resolve(Math.round(percent * 100));
+		});
+	});
+}
+
+const blockDeviceRegex = /(\/dev\/.*)p\d+/;
+export async function getStorageInfo() {
+	const fsInfo = await systeminformation.fsSize();
+	let mainFs: string | undefined;
+	let total = 0;
+	// First we find the block device which the data partition is part of
+	for (const partition of fsInfo) {
+		if (partition.mount === '/data') {
+			const match = partition.fs.match(blockDeviceRegex);
+			if (match == null) {
+				mainFs = undefined;
+			} else {
+				mainFs = match[1];
+				total = partition.size;
+			}
+			break;
+		}
+	}
+
+	if (!mainFs) {
+		return {
+			blockDevice: '',
+			storageUsed: undefined,
+			storageTotal: undefined,
+		};
+	}
+
+	let used = 0;
+	for (const partition of fsInfo) {
+		if (partition.fs.startsWith(mainFs)) {
+			used += partition.used;
+		}
+	}
+
+	return {
+		blockDevice: mainFs,
+		storageUsed: bytesToMb(used),
+		storageTotal: bytesToMb(total),
+	};
+}
+
+export async function getMemoryInformation() {
+	const mem = await systeminformation.mem();
+	return {
+		used: bytesToMb(mem.used),
+		total: bytesToMb(mem.total),
+	};
+}
+
+export async function getCpuTemp() {
+	return Math.round((await systeminformation.cpuTemperature()).main);
```

Comment:
The api will accept decimals, if we aren't going to report decimals then we should probably update the api, cc @richbayliss